staging/vhost-device-can/src/vhu_can.rs

```diff
@@ -0,0 +1,2181 @@
+        //         VIRTIO_CAN_FLAGS_FD in flags.
+        let frame = VirtioCanFrame {
+            msg_type: 0.into(),
+            can_id: 0.into(),
+            length: 40.into(),
+            reserved: 0.into(),
+            flags: CAN_FRMF_TYPE_FD.into(),
+            sdu: [0; 64],
+        };
+
+        assert_eq!(
+            vu_can_backend
+                .check_rx_frame(frame)
+                .unwrap()
+                .flags
+                .to_native()
+                & VIRTIO_CAN_FLAGS_FD,
+            VIRTIO_CAN_FLAGS_FD
+        );
+
+        // Test 3: receive frame with the same length if length is < 64
+        assert_eq!(
+            vu_can_backend.check_rx_frame(frame).unwrap().length,
+            frame.length
+        );
+
+        // Test 4: receive frame with length == 64 if length is > 64
+        let frame = VirtioCanFrame {
+            msg_type: 0.into(),
+            can_id: 0.into(),
+            length: 80.into(),
+            reserved: 0.into(),
+            flags: CAN_FRMF_TYPE_FD.into(),
+            sdu: [0; 64],
+        };
+
+        assert_eq!(vu_can_backend.check_rx_frame(frame).unwrap().length, 64);
+    }
+
+    #[test]
+    fn test_virtio_can_check_rx_canfd_vcan0() {
+        let controller =
+            CanController::new("vcan0".to_string()).expect("Could not build controller");
+        let controller = Arc::new(RwLock::new(controller));
+        let mut vu_can_backend =
+            VhostUserCanBackend::new(controller.clone()).expect("Could not build vhucan device");
+
+        // Enable VIRTIO_CAN_F_CAN_FD feature
+        vu_can_backend.acked_features((1 << VIRTIO_CAN_F_CAN_FD) | (1 << VIRTIO_CAN_F_CAN_CLASSIC));
+
+        // If VIRTIO_CAN_F_CAN_FD  and VIRTIO_CAN_F_CAN_CLASSIC are negotiated
+        // and interface is "vcan0" check if return message has
+        // VIRTIO_CAN_FLAGS_FD in flags and has been treated as CANFD frame.
+        let frame = VirtioCanFrame {
+            msg_type: 0.into(),
+            can_id: 0.into(),
+            length: 40.into(),
+            reserved: 0.into(),
+            flags: 0.into(),
+            sdu: [0; 64],
+        };
+
+        assert_eq!(
+            vu_can_backend
+                .check_rx_frame(frame)
+                .unwrap()
+                .flags
+                .to_native()
+                & VIRTIO_CAN_FLAGS_FD,
+            VIRTIO_CAN_FLAGS_FD
+        );
+
+        assert_eq!(
+            vu_can_backend.check_rx_frame(frame).unwrap().length,
+            frame.length
+        );
+    }
+
+    #[test]
+    fn test_virtio_can_rx_request() {
+        let controller =
+            CanController::new("can0".to_string()).expect("Could not build controller");
+        let controller = Arc::new(RwLock::new(controller));
+        let mut vu_can_backend =
+            VhostUserCanBackend::new(controller.clone()).expect("Could not build vhucan device");
+
+        // Artificial memory
```

Comment:
I do not think this and the other comments are useful.

Reply:
Ok most of those comments were deleted, only the ones describing the tests are kept.